### PR TITLE
Add models for reports calendar

### PR DESCRIPTION
**Pull request by arturs105**
Ref #813 

I decided to split this up into several smaller PRs for easier reviews.

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        public List<CalendarDayViewModel> Days { get; }
+            = new List<CalendarDayViewModel>();
+
+        public CalendarMonth CalendarMonth { get; }
+
+        public CalendarPageViewModel(CalendarMonth calendarMonth)
+        {
+            CalendarMonth = calendarMonth;
+
+            var firstDayOfMonth = (int)new DateTime(
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs]: This would deserve its own function as you use the almost same 5 lines again in the constructor.

File: Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+namespace Toggl.Foundation.MvvmCross.ViewModels
+{
+    public sealed class CalendarPageViewModel
+    {
+        public List<CalendarDayViewModel> Days { get; }
+            = new List<CalendarDayViewModel>();
+
+        public CalendarMonth CalendarMonth { get; }
+
+        public CalendarPageViewModel(CalendarMonth calendarMonth)
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs]: This should probably take in mind the `Toggl.Multivac.IUser.BeginningOfWeek` property. Now we assume that the user prefers Monday as the first day of the week.

File: Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            CalendarMonth = calendarMonth;
+
+            var firstDayOfMonth = (int)new DateTime(
+                calendarMonth.Year,
+                calendarMonth.Month, 1)
+                    .DayOfWeek;
+            firstDayOfMonth = convertToMondayBased(firstDayOfMonth);
+
+            //Add a few days from previous month
+            if (firstDayOfMonth != 0)
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs]: You don't need this if, you can do it the same way as with the last week.

File: Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                {
+                    Days.Add(new CalendarDayViewModel(
+                        daysInPreviousMonth - i,
+                        false));
+                }
+            }
+
+            //Add days from the target month
+            var daysInMonth = calendarMonth.DaysInMonth;
+            for (int i = 0; i < daysInMonth; i++)
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs]: Maybe you could refactor this to a separate method, which would get a range of days + the boolean and add the days? This constructor is IMO too long and the for loops are used three times.

File: Toggl.Foundation.MvvmCross/ViewModels/ReportsCalendarViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        public override void Prepare()
+        {
+            base.Prepare();
+
+            var now = timeService.CurrentDateTime;
+
+            CurrentMonth = new CalendarMonth(now.Year, now.Month);
+
+            var monthIterator
+                = new CalendarMonth(now.Year, now.Month).AddMonths(-11);
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/ReportsCalendarViewModel.cs]: Give this constant a name please.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+                await ViewModel.Initialize();
+
+                ViewModel.Months.Should().HaveCount(12);
+                var firstDateTime = now.AddMonths(-11);
+                var month = new CalendarMonth(
+                    firstDateTime.Year, firstDateTime.Month);
+                for (int i = 0; i < 12; i++, month = month.Next())
+                {
+                    ViewModel.Months[i].CalendarMonth.Should().Be(month);
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs]: Please also test that the beginning of week is the same as the user's preference.

File: Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+using Toggl.Multivac;
+
+namespace Toggl.Foundation.MvvmCross.ViewModels
+{
+    public sealed class CalendarPageViewModel
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/CalendarPageViewModel.cs]: There are no tests for this class and I think this should be tested very carefully.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs (hunk 6 above)
Comment (arturs105) [reply Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs]: There isn't an easy way to test this since `CalendarPageViewModel` doesn't publicly expose the beginning of week. Though, i think, this will be covered in `CalendarPageViewModel` tests.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/CalendarPageViewModelTests.cs [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                    .Skip(calendarMonth.DaysInMonth);
+
+                if (expectedDayCount == 0)
+                    daysFromNextMonth.Should().BeEmpty();
+                else
+                    daysFromNextMonth
+                        .Should()
+                        .OnlyContain(day => !day.IsInCurrentMonth);
+            }
+        }
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/CalendarPageViewModelTests.cs]: Please add one more test for me - that the number of days % 7 is always zero.
